### PR TITLE
Added Inventory and Item pickup

### PR DESCRIPTION
**Pull request by Phezygg**

### CODE REVIEW

File: engine/maptile.py [hunk 1]
```diff
@@ -38,3 +38,11 @@ def reveal_npcs(self):
     
     def greet_npcs(self):
         return [npc.greet() for npc in self.local_npcs]
+
+    def pickupitem(self, description):
```
Comment (sogh) [inline engine/maptile.py]: Let's try to either use underscores or CamelCase in function names to keep up readability. Which do you prefer?
PickUpItem, or pick_up_item ?

File: engine/maptile.py [hunk 2]
```diff
@@ -38,3 +38,11 @@ def reveal_npcs(self):
...
+        item_index = 0
+        for item in self.local_items:
+            if item.description == description:
+                return self.local_items.pop(item_index)
+            item_index += 1
+        return "You cannot get that."
```
Comment (sogh) [inline engine/maptile.py]: In general this is something unique about Python - you can return multiple different types from a single function. However, it's a confusing practice. I'd recommend you return None here if you can't find the item, and have the caller of this function generate this string when you return None.

File: engine/player.py [hunk 3]
```diff
@@ -34,4 +35,10 @@ def move(self,direction):
         elif direction.lower() == 'west' or direction.lower() == 'w':
             self.__position.x = self.__position.x - 1
         else:
-            print('Invalid Direction')
\ No newline at end of file
+            print('Invalid Direction')
+
+    def additemtoinv(self, new_item):
```
Comment (sogh) [inline engine/player.py]: same here, let's keep function/method names more readable with _ or CamelCase

File: mygame.py [hunk 4]
```diff
@@ -67,6 +70,16 @@ def Step(self):
             elif lowcmd in self.npc_commands:
                 for greeting in self.worldmap.greetnpcs(self.player1.position.x, self.player1.position.y):
                     self.graphics.RenderText(greeting)
+            elif lowcmd in self.item_commands:
+                item_get = input("What will you pick up?")
+                item = self.worldmap.pickupitems(self.player1.position.x, self.player1.position.y, item_get)
+                if item == "You cannot get that.":
```
Comment (sogh) [inline mygame.py]: so here, if pickupitems returns None, this if statement becomes cleaner:

if item:
  self.player1.additemtoinv(item)
else:
  print("You cannot get that.")  

now you dont repeat that string, and skip a string comparison operation.

File: mygame.py [hunk 5]
```diff
@@ -67,6 +70,16 @@ def Step(self):
...
+                    print("You cannot get that.")
+                else:
+                    self.player1.additemtoinv(item)
+            elif lowcmd in self.inventory_commands:
```
Comment (sogh) [inline mygame.py:80]: I see what you mean here, maybe you can at least say something like 
RenderText("You are carrying:")
for ...
   RenderText("    " + item)

And we can brainstorm a better graphics engine.

**Review by sogh — CHANGES_REQUESTED** [review]

File: mygame.py (hunk 5 above)
Comment (Phezygg) [reply mygame.py:80]: The issue I think is that it shows you everything each passthrough.  Whereas if I type inventory I would just want to see that, or if I type look I would just want to see the results.

**Review by sogh — APPROVED** [review]

**Pull request merged**